src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/ContentPickerFieldSettingsDriver.cs

```diff
@@ -59,12 +65,30 @@ public override async Task<IDisplayResult> UpdateAsync(ContentPartFieldDefinitio
                         break;
                 }
 
-                context.Builder.WithSettings(settings);
+                var isValid = ValidateTextPatterns(context, model);
+
+                if (isValid)
+                {
+                    context.Builder.WithSettings(settings);
+                }
             }
 
             return Edit(partFieldDefinition);
         }
 
+        private bool ValidateTextPatterns(UpdatePartFieldEditorContext context, ContentPickerFieldSettingsViewModel model)
+        {
+            bool isValid = true;
+
+            if (!string.IsNullOrEmpty(model.TitlePattern) && !_templateManager.Validate(model.TitlePattern, out var titleErrors))
+            {
+                context.Updater.ModelState.AddModelError(nameof(model.TitlePattern), S["TitlePattern doesn't contain a valid Liquid expression. Details: {0}", string.Join(" ", titleErrors)]);
+                isValid = false;
```

Comment:
Please re-label `S["TitlePattern doesn't contain a valid Liquid expression. Details: {0}"` to `S["Title Pattern does not contain a valid Liquid expression. Details: {0}"`

The `return false;`